modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -701,43 +702,39 @@ private CompletableFuture<Void> waitForMetadataCompleteness(long ts) {
 
     private CompletableFuture<Boolean> onZoneReplicaCreated(LocalPartitionReplicaEventParameters parameters) {
         if (!PartitionReplicaLifecycleManager.ENABLED) {
-            return completedFuture(false);
+            return falseCompletedFuture();
         }
 
         return inBusyLockAsync(busyLock, () -> readyToProcessTableStarts
                 .thenCompose(v -> {
-                    Set<TableImpl> zoneTables = zoneTables(parameters.zonePartitionId().zoneId());
+                    ZonePartitionId zonePartitionId = parameters.zonePartitionId();
 
-                    PartitionSet singlePartitionIdSet = PartitionSet.of(parameters.zonePartitionId().partitionId());
+                    Set<TableImpl> zoneTables = zoneTables(zonePartitionId.zoneId());
 
-                    CompletableFuture<?>[] futures = zoneTables.stream()
-                            .map(tbl -> {
-                                CompletableFuture<Void> createStoragesFuture = runAsync(
```

Comment:
To reviewer, this change is needed, because there was a bug: `getOrCreatePartitionStorages` returns a `CompletableFuture`, which we wrapped in a `runAsync` having a double wrapped future, which means we didn't wait for the nested future to complete